> > Is irrlicht a bundled dependency built within the build of voxeland?
> 
> No. If I see it correctly voxeland uses system irrlicht.

Then if irrlicht was the failing build, the build log wouldn't have looked like this:

```
[ 99%] Linking CXX executable ../../bin/voxelands-server
[ 99%] Built target voxelands-server
[100%] Linking CXX executable ../../bin/voxelands
/nix/store/liq5wrz3yprc9623bb9lc90qv610gsd8-binutils-2.31.1/bin/ld: /nix/store/ci25cir0pkj8h2dmsaxppdcyqif2m0cb-irrlicht-1.8.4/lib/libIrrlicht.so: undefined reference to `png_init_filter_functions_neon'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/voxelands.dir/build.make:1370: ../bin/voxelands] Error 1
make[1]: *** [CMakeFiles/Makefile2:186: src/CMakeFiles/voxelands.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
error: --- Error --- nix-build
error: --- Error --- nix-daemon
builder for '/nix/store/9lzw62zdgkzq8jb25xs9yq22ar793q81-voxelands-1704.00.drv' failed with exit code 2; last 10 log lines:
    414 |   for (u8 i = 0; i < 8; i++) {
        |   ^~~
  [ 99%] Linking CXX executable ../../bin/voxelands-server
  [ 99%] Built target voxelands-server
  [100%] Linking CXX executable ../../bin/voxelands
  /nix/store/liq5wrz3yprc9623bb9lc90qv610gsd8-binutils-2.31.1/bin/ld: /nix/store/ci25cir0pkj8h2dmsaxppdcyqif2m0cb-irrlicht-1.8.4/lib/libIrrlicht.so: undefined reference to `png_init_filter_functions_neon'
  collect2: error: ld returned 1 exit status
```

As you can see, it's voxelands' binaries that fail to link, not irrlicht's. Whatever the reason is, we should mark it as broken for aarch64, ideally with a comment saying the build expects neon symbols or something like that.

--- a/pkgs/games/voxelands/default.nix
+++ b/pkgs/games/voxelands/default.nix
@@ -1,13 +1,30 @@
-{ stdenv, fetchurl, cmake, irrlicht, libpng, bzip2, sqlite
-, libjpeg, libXxf86vm, libGLU, libGL, openal, libvorbis, xlibsWrapper, pkgconfig }:
+{ stdenv
+, fetchFromGitLab
+, bzip2
+, cmake
+, expat
+, irrlicht
+, libGL
+, libGLU
+, libXxf86vm
+, libjpeg
+, libpng
+, libvorbis
+, openal
+, pkg-config
+, sqlite
+, xlibsWrapper
+}:
 
 stdenv.mkDerivation rec {
   pname = "voxelands";
-  version = "1512.00";
+  version = "1704.00";
 
-  src = fetchurl {
-    url = "http://voxelands.com/downloads/${pname}-${version}-src.tar.bz2";
-    sha256 = "0bims0y0nyviv2f2nxfj37s3258cjbfp9xd97najz0yylnk3qdfw";
+  src = fetchFromGitLab {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "0yj9z9nygpn0z63y739v72l3kg81wd71xgix5k045vfzhqsam5m0";
   };
 
   cmakeFlags = [
@@ -16,16 +33,32 @@ stdenv.mkDerivation rec {
     "-DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG"
   ];
 
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
   buildInputs = [
-    cmake irrlicht libpng bzip2 libjpeg sqlite
-    libXxf86vm libGLU libGL openal libvorbis xlibsWrapper pkgconfig
+    bzip2
+    expat
+    irrlicht
+    libGL
+    libGLU
+    libXxf86vm
+    libjpeg
+    libpng
+    libvorbis
+    openal
+    sqlite
+    xlibsWrapper
   ];
 
   meta = with stdenv.lib; {
-    homepage = "http://voxelands.com/";
+    homepage = "https://voxelands.net/";
     description = "Infinite-world block sandbox game based on Minetest";
     license = licenses.gpl3Plus;
     platforms = platforms.linux;
     maintainers = with maintainers; [ ];
+    broken = stdenv.isAarch64;  # build fails with "libIrrlicht.so: undefined reference to `png_init_filter_functions_neon'"
   };
 }

